elastic/kibana
# [Fleet] Allow package to specify cluster privileges

## Summary

Resolve #114686
Related to https://github.com/elastic/package-spec/pull/226
Depends on https://github.com/elastic/package-registry/pull/750

Allow packages to specify cluster privileges:
* packages can specify `elasticsearch.privileges.cluster: ["monitor"]` in the package manifest.yml this will be returned by the package registry API
* When adding|editing|updating a package policy we save that info on the package policy saved object
* When generating the agent policy we use that to generate the correct privileges
 

## How to tests

I added basic unit test to ensure we generate the correct privileges


This PR is a little hard to test locally as there is a lot of moving parts, this is how I tested it my self:
* I am running the package registry locally based on that PR https://github.com/elastic/package-registry/pull/750
* I created a new version of the system integrations locally with `elasticsearch.privileges.cluster: ["monitor_ml"]` in the manifest.yml
* Than I added the system integration to a policy enrolled agents and checked in `.fleet-agents` for the default api key to check if the API key has the correct privileges
